frontend/src/reducers/reducer.js

```diff
@@ -26,28 +23,19 @@ export const reducer = (state, action) => {
             return { ...state, error: payload, loading: false }
 
         case ADD_TO_WISHLIST:
-            return { ...state, wishlist: [...state.wishlist, payload] }
+            return { ...state, wishlist: payload }
 
         case REMOVE_FROM_WISHLIST:
-            const updatedWishlist = state.wishlist.filter((x) => x.id !== payload);
-            return { ...state, wishlist: updatedWishlist }
+            return { ...state, wishlist: payload }
```

Comment:
Since both of these conditions are performing the same action, it can be refactored as:
```javascript
case ADD_TO_WISHLIST:
case REMOVE_FROM_WISHLIST:
return {...state, wishlist: payload};
```

Same can be followed for other cases as well (ex: type for cart actions)

Reply:
done